cmd/launcher/svc_config_windows.go

```diff
@@ -213,10 +224,28 @@ func setRecoveryActions(ctx context.Context, logger *slog.Logger, service *mgr.S
 		},
 	}
 
+	// If the recovery actions are already set, we don't need to do anything
+	if recoveryActionsAreSet(curRecoveryActions, recoveryActions) {
+		return
+	}
+
 	if err := service.SetRecoveryActions(recoveryActions, 24*60*60); err != nil { // 24 hours
 		logger.Log(ctx, slog.LevelError,
 			"setting RecoveryActions",
 			"err", err,
 		)
 	}
 }
+
+// recoveryActionsAreSet checks if the current recovery actions are set to the desired recovery actions
+func recoveryActionsAreSet(curRecoveryActions []mgr.RecoveryAction, recoveryActions []mgr.RecoveryAction) bool {
+	if curRecoveryActions == nil || len(curRecoveryActions) != len(recoveryActions) {
+		return false
+	}
+	for i := range curRecoveryActions {
+		if curRecoveryActions[i].Type != recoveryActions[i].Type && curRecoveryActions[i].Delay != recoveryActions[i].Delay {
```

Comment:
Should this be an `||`

I might write it as two if blocks, because compound negative statements are confusing

Reply:
An OR would make more sense if the RecoveryActions differed in Type or Delay. I went with an AND here since they all have the same type and delay. Is that thought process flawed?

Reply:
I think we'd want to overwrite if either the type or the delay didn't match. This is what the split statements do, and is readable.

I think if you combine them, it would be an `||`, otherwise the delay matching, but the type mismatching would _not_ return `false`. But I always get tangled in compound negation.